lib/Service/UserMigrationService.php

```diff
@@ -94,12 +99,73 @@ public function __construct(
 		$this->exportMapper = $exportMapper;
 		$this->importMapper = $importMapper;
 		$this->jobList = $jobList;
+		$this->internalCache = new CappedMemoryCache();
 
 		$this->mandatory = true;
 	}
 
 	/**
 	 * @param ?string[] $filteredMigratorList If not null, only these migrators will run. If empty only the main account data will be exported.
+	 *
+	 * @return int Estimated size in KiB
+	 *
+	 * @throws UserMigrationException
+	 */
+	public function estimateExportSize(IUser $user, ?array $filteredMigratorList = null): int {
+		// 4MiB for base user data
+		$size = 4096;
+
+		foreach ($this->getMigrators() as $migrator) {
+			if ($filteredMigratorList !== null && !in_array($migrator->getId(), $filteredMigratorList)) {
+				continue;
+			}
+			$cacheKey = $user->getUID() . '::' . $migrator->getId();
+			if ($this->internalCache->hasKey($cacheKey)) {
+				$size += $this->internalCache->get($cacheKey);
+				continue;
+			}
+			if ($migrator instanceof ISizeEstimationMigrator) {
+				try {
+					$migratorSize = $migrator->getEstimatedExportSize($user);
+				} catch (Throwable $e) {
+					throw new UserMigrationException('Could not estimate export size for ' . $migrator->getDisplayName(), 0, $e);
+				}
+				$this->internalCache->set($cacheKey, $migratorSize);
+				$size += $migratorSize;
+			}
+		}
+
+		return $size;
+	}
+
+	/**
+	 * @param ?string[] $filteredMigratorList If not null, only these migrators will run. If empty only the main account data will be exported.
+	 *
+	 * @throws NotExportableException
+	 */
+	public function checkExportability(IUser $user, ?array $filteredMigratorList = null): void {
+		try {
+			$userFolder = $this->root->getUserFolder($user->getUID());
+			$freeSpace = (int)ceil($userFolder->getFreeSpace() / 1024);
```

Comment:
You need to add the previous export file size to the freeSpace value, as it will be overwritten by the export.